app/controllers/candidate_interface/work_history/edit_controller.rb

```diff
@@ -28,20 +14,27 @@ def create
       end
     end
 
-    def show
-      @application_form = current_candidate.current_application
+    def edit
+      work_experience = current_candidate.current_application
+        .application_work_experiences.find(edit_params[:id])
```

Comment:
as before, I am not sure it's good to call the params, with a name of the action, it's clearer to call them with an explicit name (a.i. `application_params`, `candidate_params`, ....)